Migrate backends/apple to the new namespace

dbort opened the pull request:
Summary: Move the Apple backends out of the `torch::` namespace, and update to avoid using the `torch::` or `exec_aten::` namespaces.

Differential Revision: D63908530

cccclai reviewed (approved):
Looks good if CI passes.

DenisVieriu97 reviewed (approved):
looks good

dbort committed:
Migrate backends/apple/mps to the new namespace (#5883)

Summary:

Move the MPS backend out of the `torch::` namespace, and update to avoid using the `torch::` or `exec_aten::` namespaces.

Reviewed By: cccclai

Differential Revision: D63908530

dbort commented:
@pytorchbot cherry-pick --onto release/0.4 -c fixnewfeature
